test/unit_test/character_sheet_unit_tests/FocusesSetUnitTests.java

```diff
@@ -0,0 +1,86 @@
+package unit_test.character_sheet_unit_tests;
+
+import static org.junit.Assert.*;
+
+import java.security.InvalidParameterException;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import rpg_database.character_sheet.CharacterSheet;
+import rpg_database.character_sheet.Fields;
+import rpg_database.character_sheet.Focuses;
+import rpg_database.character_sheet.Focus;
+import rpg_database.character_sheet.FocusesSet;
+
+public class FocusesSetUnitTests {
+
+	@Rule
+	public ExpectedException thrown = ExpectedException.none();
+
+	@Test
+	public void expectException_SetFocusesMalformedInput() {
+		expectExceptionWithMessage(InvalidParameterException.class,
+				"class java.lang.String value is not an instance of class rpg_database.character_sheet.Focuses");
+		final String malformedInput = "MALFORMED INPUT";
+		CharacterSheet characterSheet = new CharacterSheet("CharacterSheet");
+		characterSheet.setData(Fields.FOCUSES, malformedInput);
+	}
+
+	// functional tests
+	@Test
+	public void testSetFocusWithFocusesSet() {

```

Comment:
Ezt a funkcionalitást, hogy ```new FocusesSet(new Focus(Focuses.ETIQUETTE));``` ne támogasd. Ha Focuses enumokból kitalálja hogy minek kell benne lennie és milyen értékkel akkor felesleges Focus objektumokat átadogatni neki. Azt szeretnénk hogy ő a Focus objektumokat belül hozza létre és ez számunkra kívülről ne legyen látható.

Ergo ez a teszt kuka. Az egyel lejjebbinek kell helyettesítenie.